test/components/agents/test_agent.py

```diff
@@ -746,7 +746,7 @@ def test_run(self, weather_tool):
         chat_generator = OpenAIChatGenerator(model="gpt-4o-mini")
         agent = Agent(chat_generator=chat_generator, tools=[weather_tool], max_agent_steps=3)
         agent.warm_up()
-        response = agent.run([ChatMessage.from_user("What is the weather in Berlin?")])
+        response = agent.run([ChatMessage.from_user("What is the weather in Berlin?")], generation_kwargs={"temperature": 0.0})
```

Comment:
Could we add a new unit test instead that resuses `chat_generator = MockChatGeneratorWithoutRunAsync()` and `chat_generator.run.assert_called_once_with(messages=..., tools=..., generation_kwargs=...)` so we can check that the `generation_kwargs` are being properly propogated?

Reply:
fixed